File: data-structures/dynamic-ring-buffer/interfaces/dynamic-ring-buffer.ts
export interface DynamicRingBuffer<T> {
  length(): number;
  capacity(): number;
  isEmpty(): boolean;
  isFull(): boolean;
  front(): T | undefined;
  rear(): T | undefined;
  enqueue(value: T): boolean;
  dequeue(): T | undefined;
  clear(): void;
  toArray(): T[];
}

File: data-structures/dynamic-ring-buffer/interfaces/dynamic-ring-buffer.go
package interfaces

type DynamicRingBuffer[T any] interface {
	Length() int
	Capacity() int
	IsEmpty() bool
	IsFull() bool
	Front() T
	Rear() T
	Enqueue(value T) bool
	Dequeue() T
	Clear()
	ToSlice() []T
}
